src/python/gem5/prebuilt/hifiveunmatched/hifive_cache.py

```diff
@@ -0,0 +1,185 @@
+        the PrivateL1PrivateL2 CacheHierarchy.
+        :returns: The default memory bus for the PrivateL1PrivateL2
+        CacheHierarchy.
+        :rtype: SystemXBar
+        """
+        membus = SystemXBar(width=64)
+        membus.badaddr_responder = BadAddr()
+        membus.default = membus.badaddr_responder.pio
+        return membus
+
+    def __init__(
+        self,
+        l1d_size: str,
+        l1i_size: str,
+        l2_size: str,
+        membus: BaseXBar = _get_default_membus.__func__(),
+    ) -> None:
+        """
+        :param l1d_size: The size of the L1 Data Cache (e.g., "32kB").
+        :type l1d_size: str
+        :param  l1i_size: The size of the L1 Instruction Cache (e.g., "32kB").
+        :type l1i_size: str
+        :param l2_size: The size of the L2 Cache (e.g., "256kB").
+        :type l2_size: str
+        :param membus: The memory bus. This parameter is optional parameter and
+        will default to a 64 bit width SystemXBar is not specified.
+        :type membus: BaseXBar
+        """
+
+        AbstractClassicCacheHierarchy.__init__(self=self)
+        AbstractTwoLevelCacheHierarchy.__init__(
+            self,
+            l1i_size=l1i_size,
+            l1i_assoc=4,
+            l1d_size=l1d_size,
+            l1d_assoc=8,
+            l2_size=l2_size,
+            l2_assoc=16,
+        )
+
+        self.membus = membus
+
+    @overrides(AbstractClassicCacheHierarchy)
+    def get_mem_side_port(self) -> Port:
+        return self.membus.mem_side_ports
+
+    @overrides(AbstractClassicCacheHierarchy)
+    def get_cpu_side_port(self) -> Port:
+        return self.membus.cpu_side_ports
+
+    @overrides(AbstractCacheHierarchy)
+    def incorporate_cache(self, board: AbstractBoard) -> None:
+
+        # Set up the system port for functional access from the simulator.
+        board.connect_system_port(self.membus.cpu_side_ports)
+
+        for cntr in board.get_memory().get_memory_controllers():
+            cntr.port = self.membus.mem_side_ports
+
+        self.l1icaches = [
+            L1ICache(size=self._l1i_size)
+            for i in range(board.get_processor().get_num_cores())
+        ]
+        self.l1dcaches = [
+            L1DCache(size=self._l1d_size)
+            for i in range(board.get_processor().get_num_cores())
+        ]
+        self.l2buses = [
+            L2XBar() for i in range(board.get_processor().get_num_cores())
+        ]
+        self.l2caches = [
+            L2Cache(size=self._l2_size)
+            for i in range(board.get_processor().get_num_cores())
+        ]
+        # ITLB Page walk caches
+        self.iptw_caches = [
+            MMUCache(size="4KiB")
+            for _ in range(board.get_processor().get_num_cores())
+        ]
+        # DTLB Page walk caches
+        self.dptw_caches = [
+            MMUCache(size="4KiB")
+            for _ in range(board.get_processor().get_num_cores())
+        ]
```

Comment:
Out of curiosity, are these listed in the spec?

Reply:
Yeah, I might be mistaken in interpreting it, but on page 55 (section 4.7.7) of the datasheet, there is a mention of the level 2 TLB pagesize being 4 KiB. On writing this, I realize the importance of citing the page number and section from where we picked up the data :)

Reply:
A two level TLB and a PWC are two different things. The current implementation is good with me, but we should make a note of the discrepancy.